cli/src/swam/cli/Main.scala

```diff
@@ -318,7 +328,38 @@ object Main extends CommandIOApp(name = "swam-cli", header = "Swam from the comm
                 res <- engine.validate(module).attempt
                 _ <- res.fold(t => logger.error("Module is invalid", t), _ => logger.info("Module is valid"))
               } yield ExitCode.Success
+            case Infer(file, wat, wasi, func_name) =>
+              for {
+                engine <- Engine[IO](blocker)
+                tcompiler <- swam.text.Compiler[IO](blocker)
+                module = if (wat) tcompiler.stream(file, false, blocker) else engine.sections(file, blocker)
 
+                compiled <- engine.compile(module)
+                _ <- IO(compiled.functions.zipWithIndex.map {
+                        case (CompiledFunction(typeIndex, tpe, locals, code), fidx) => 
+                          val functionName =
+                          compiled.names.flatMap(_.subsections.collectFirstSome {
+                            case FunctionNames(names) =>
+                              names.get(typeIndex)
+                            case _ =>
+                              None
+                        })
+                        functionName match {
+                          case Some(x) => {
+                            if(x == func_name){
+                              println(s"This is function name : $func_name")
```

Comment:
This line is redundant, we already know the function name :). Remove it

Reply:
Removed line in commit 324334e